src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2677,17 +2678,18 @@ private void PushOne()
                 }
             }
 
-            private void HolderCompleted(Holder<TOut> holder)
+            private void HolderCompleted(Holder<TOut> holder, Result<TOut> result)
```

Comment:
This is the real fix - we now pass in both the `Holder<TOut>` and the `Result<TOut>` together into the StageActor's context and perform the assignment there, where it's guaranteed to be thread-safe, rather than doing it half-in / half-out like before.